service/history/nDCEventsReapplier.go

```diff
@@ -108,5 +108,18 @@ func (r *nDCEventsReapplierImpl) reapplyEvents(
 		deDupResource := definition.NewEventReappliedID(runID, event.GetEventId(), event.GetVersion())
 		msBuilder.UpdateDuplicatedResource(deDupResource)
 	}
+
+	// After reapply event, checking if we should schedule a workflow task
+	// Do not create workflow task when the workflow is cron and the cron has not been started yet
+	if msBuilder.GetExecutionInfo().CronSchedule != "" && !msBuilder.HasProcessedOrPendingWorkflowTask() {
```

Comment:
instead of checking for cronSchedule, you should check firstWorkflowTaskBackoff

Reply:
If cron has the first wf task, then we should continue schedule a task. Is this correct?

Reply:
We also need to take care of workflow retry backoff case I think.